storage/scratchpad/src/sparse_merkle/sparse_merkle_test.rs

```diff
@@ -99,14 +106,24 @@ fn test_construct_subtree_with_new_leaf_override_existing_leaf() {
     let subtree = SparseMerkleTree::construct_subtree_with_new_leaf(
         key,
         new_blob.clone(),
-        existing_leaf,
+        existing_leaf.clone(),
         key,
         /* distance_from_root_to_existing_leaf = */ 3,
     );
 
     let new_blob_hash = new_blob.hash();
     let root_hash = hash_leaf(key, new_blob_hash);
     assert_eq!(subtree.hash(), root_hash);
+
+    // batch version api
+    let subtree_batch = SparseMerkleTree::batch_construct_subtree_with_existing_leaf(
```

Comment:
The "non-batch" versions of APIs should be removed directly IMHO.

Reply:
you mean per-txn non-batch version, right?